### ISSUE TITLE
Output color variables as JSON and JavaScript

### ISSUE THREAD

**Issue by gneatgeek** [cerner/terra-legacy-theme#5]
In order to pull the color values into JavaScript for coloring objects and visuals, I'd like to export the Sass colors as a JSON object and have a build task to wrap it in an ES6 module.

**Comment by bunnyLord** [comment]
I can try this one. What unit testing framework suggestions do you have since there is none used currently? 
Also can this be used https://www.npmjs.com/package/scss-to-json ? It has a MIT license.

**Comment by bjankord** [comment]
Hey @bunnyLord, I think the [scss-to-json](https://www.npmjs.com/package/scss-to-json) lib would be great for this issue.  The scss-to-json lib has [tests](https://github.com/ryanbahniuk/scss-to-json/tree/master/test) so I do not think the work for this issue needs any tests at this time.

**Comment by bunnyLord** [comment]
Sorry I've been sick this weekend. Will try to get it done in the next couple of days.

**Comment by gneatgeek** [comment]
@bunnyLord, no rush.  I hope you feel better!

**Comment by bunnyLord** [comment]
Is there another project that consumes this I can use to make sure everything works as expected? Because there are no tests here. Only thing I can run is the linters. @bjankord @gneatgeek

**Comment by bjankord** [comment]
@bunnyLord I don't believe we have any projects that are consuming this outside of the terra components. I guess we could test that the generated file is valid JSON. For now, we'll have to manually verify the output though. 

We talked about generating documentation from the JSON file at some point in the future to view all the colors in the browser similar to https://material.google.com/style/color.html#color-color-palette. With that, we could set up visual regression tests to ensure the JSON outputs the expected colors. Until then, we don't have an automated way to test the generated output.

**Comment by bunnyLord** [comment]
@bjankord would "With that, we could set up visual regression tests to ensure the JSON outputs the expected colors." be outside the scope of this task? How much work would that be
The thing is the library I mentioned is going to make the SCSS colors available as variables in memory, not necessarily output them to a file. To do the latter, I think we need to use grunt-scss-to-json. So to output the vars to a JSON file I can access directly (not through JS script) I would have to add grunt config that will likely be removed when proper visual tests are added. 
So I think adding the visual tests now would be a better option unless I misunderstood what you meant by your suggestion to check if the JSON file i valid and the info below.
From scss-to-json npm page: 
"This package is a function that will make this conversion for you in memory. If you want to output JSON to your file system, you should use the associated grunt-scss-to-json package."

**Comment by bjankord** [comment]
> "With that, we could set up visual regression tests to ensure the JSON outputs the expected colors." be outside the scope of this task?

Yes, I would consider that out of scope for this issue. 

I don't think we want to take on grunt as a dependency for this project. We've been trying to stick with npm-scripts for any build tooling. Knowing the scss-to-json lib makes the conversion in memory and doesn't actually output to a file, I'd recommend looking at writing a node.js script and use [node's fs module|https://nodejs.org/api/fs.html] to write conversion from scss-to-json to a file. You could then run that view npm-scripts with a script like `build-json: "node build-json.js"` where build-json.js is the node.js script. I think the grunt-scss-to-json module offers an [implementation](https://github.com/ryanbahniuk/grunt-scss-to-json/blob/master/tasks/scss_to_json.js#L45-L46) we could work out in a node.js script. 

I'll reach out to the rest of our team to see if we can get some thoughts together on how to go about testing this. My thought is validating the json values are the correct colors is something we'll need to do manually and enhance later.

**Comment by bjankord** [comment]
Here is a quick go at what I'm thinking to generate a JSON file:

```
const scssToJson = require('scss-to-json');
const fs = require('fs');

const colors = JSON.stringify(scssToJson('src/terra/legacy-theme/_palette.scss'));

fs.writeFile('src/pallette.json', colors, (err) => {
  if (err) {
    return console.log(err);
  }
});
```

**Comment by gneatgeek** [comment]
In regards to testing, I would check that the file:
- is valid JSON
- is saved to disk
- contains one or two of the colors we expect it to have

`scss-to-json` is already tested so I don't feel the need to ensure that it works.

**Comment by bunnyLord** [comment]
When asking about tests, I pretty much meant testing behaviors you outlined above @gneatgeek, not testing scss-to-json. Looks like we do need some tests after all to verify that whatever I PR actually works as expected.

We can have build-json and test-build-json where test-build-json will contains the tests @gneatgeek outlined above. Does that sound good?

**Comment by bjankord** [comment]
@bunnyLord I was talking with @gneatgeek about testing for this and we are currently discussing our overall testing strategy. We have some decisions we need to land on around the testing framework we use and how we integrate that into our build tooling. Since we havn't landed those decisions yet, we are planning to verify this story via exploratory testing. Once we land on a testing strategy, we'll come back through and add tests for this.  For now, don't worry about writing tests for this functionality. I'll log an issue so we remember to come back and add tests for this feature.
